### ISSUE TITLE
Bar Graph Item does not update x range if all values bars coordinates are below 1.

### ISSUE THREAD

**Issue by Flowake** [pyqtgraph/pyqtgraph#2582]
### Short description
When plotting a bar graph item, if all the edges of the bars are below 1, the x range will not update and stay at 1.
For example, if my data bars are between 0 and 0.1, the x range will remain from 0 to 1.
Using the built-in right click "view-all" does not rescale the graph either.

### Code to reproduce

```python
import numpy as np

import pyqtgraph as pg

win = pg.plot()

x = np.linspace(0,0.1,10)
y1 = np.sin(x)

bg1 = pg.BarGraphItem(x0=x[:-1], x1=x[1:], height=y1, brush='r')

win.addItem(bg1)
# win.setXRange(min(x), max(x))

if __name__ == '__main__':
    pg.exec()
```

### Expected behavior
The x range should update.
The "view all" option should rescale the graph properly.

### Real behavior
The x range does not update.

### Tested environment(s)

 * PyQtGraph version: 0.13.1
 * Qt Python binding: PySide6 6.4.0
 * Python version: 3.10
 * NumPy version: 1.23.4
 * Operating system: Linux Ubuntu 22.04
 * Installation method: pip

### Additional context
Adding the following line after adding the item to the plot properly rescale the x axis.

```python
win.setXRange(min(x), max(x))
```

**Comment by pijyoi** [comment]
This issue would be fixed by #2565. If you are able to, could you give it a try?

**Comment by j9ac9k** [comment]
going to close this as fixed by #2565 

Thanks for the report @Flowake  and thanks for the fix @pijyoi
